src/base/cub_ddl_log.c

```diff
@@ -0,0 +1,1047 @@
+      FREE_MEM (ddl_audit_handle->err_msg);
+      FREE_MEM (ddl_audit_handle);
+    }
+}
+
+void
+cub_ddl_log_app_name (const char *app_name)
+{
+  if (ddl_audit_handle == NULL || app_name == NULL)
+    {
+      return;
+    }
+
+  snprintf (ddl_audit_handle->app_name, sizeof (ddl_audit_handle->app_name), app_name);
+}
+
+void
+cub_ddl_log_db_name (const char *db_name)
+{
+  char *pstr = NULL;
+  if (ddl_audit_handle == NULL || db_name == NULL)
+    {
+      return;
+    }
+
+  snprintf (ddl_audit_handle->db_name, sizeof (ddl_audit_handle->db_name), db_name);
+
+  pstr = (char *) strchr (ddl_audit_handle->db_name, '@');
+  if (pstr != NULL)
+    {
+      *pstr = '\0';
+    }
+}
+
+void
+cub_ddl_log_user_name (const char *user_name)
+{
+  if (ddl_audit_handle == NULL || user_name == NULL)
+    {
+      return;
+    }
+
+  snprintf (ddl_audit_handle->user_name, sizeof (ddl_audit_handle->user_name), user_name);
+}
+
+void
+cub_ddl_log_ip (const char *ip_addr)
+{
+  if (ddl_audit_handle == NULL || ip_addr == NULL)
+    {
+      return;
+    }
+
+  snprintf (ddl_audit_handle->ip_addr, sizeof (ddl_audit_handle->ip_addr), ip_addr);
+}
+
+void
+cub_ddl_log_pid (const int pid)
+{
+  if (ddl_audit_handle == NULL)
+    {
+      return;
+    }
+
+  ddl_audit_handle->pid = pid;
+}
+
+void
+cub_ddl_log_br_name (const char *br_name)
+{
+  if (ddl_audit_handle == NULL || br_name == NULL)
+    {
+      return;
+    }
+  snprintf (ddl_audit_handle->br_name, 64, br_name);
+}
+
+void
+cub_ddl_log_br_index (const int index)
+{
+  if (ddl_audit_handle == NULL)
+    {
+      return;
+    }
+  ddl_audit_handle->br_index = index;
+}
+
+void
+cub_ddl_log_sql_text (char *sql_text)
+{
+  if (ddl_audit_handle == NULL || sql_text == NULL)
+    {
+      return;
+    }
+
+  if (ddl_audit_handle->sql_text != NULL)
+    {
+      FREE_MEM (ddl_audit_handle->sql_text);
+    }
+
+  ALLOC_COPY (ddl_audit_handle->sql_text, sql_text);
+}
+
+
+void
+cub_ddl_log_sql_text (char *sql_text, int len)
+{
+  if (ddl_audit_handle == NULL || sql_text == NULL || len < 0)
+    {
+      return;
+    }
+
+  if (ddl_audit_handle->sql_text != NULL)
+    {
+      FREE_MEM (ddl_audit_handle->sql_text);
+    }
+
+  ddl_audit_handle->sql_text = (char *) MALLOC (len);
+
+  if (ddl_audit_handle->sql_text != NULL)
+    {
+      strncpy (ddl_audit_handle->sql_text, sql_text, len);
+      ddl_audit_handle->sql_text[len - 1] = '\0';
+    }
+}
+
+
+void
+cub_ddl_log_stmt_type (int stmt_type)
+{
+  if (ddl_audit_handle == NULL)
+    {
+      return;
+    }
+
+  ddl_audit_handle->stmt_type = stmt_type;
+}
+
+void
+cub_ddl_log_loaddb_file_type (char file_type)
+{
+  if (ddl_audit_handle == NULL)
+    {
+      return;
+    }
+
+  ddl_audit_handle->loaddb_file_type = file_type;
+}
+
+void
+cub_ddl_log_file_name (const char *file_name)
+{
+  if (ddl_audit_handle == NULL || file_name == NULL)
+    {
+      return;
+    }
+
+  strncpy (ddl_audit_handle->file_name, file_name, PATH_MAX);
```

Comment:
I know you define '`#define DDL_LOG_PATH_MAX PATH_MAX`', but I think

`strncpy (ddl_audit_handle->file_name, file_name, DDL_LOG_PATH_MAX);`

is better. Because you define file_name [DDL_LOG_PATH_MAX] in struct definition.
Or you could remove, DDL_LOG_PATH_MAX and use PATH_MAX instead in
t_ddl_audit_handle structure.

Reply:
Thank you, i fixed it.